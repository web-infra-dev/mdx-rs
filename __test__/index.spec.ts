import { readFileSync } from "fs";
import { describe, test, expect } from "vitest";
import path from "path";
import prettier from "prettier";

import { compile } from "../index.js";

const formatHTML = (html: string) => prettier.format(html, { parser: "html" });

describe("compile", () => {
  test("should render container type correctly", async (t) => {
    let { code: result, html } = await compile({
      value: readFileSync(path.join(__dirname, "./container-type.md"), "utf8"),
      filepath: "xxx.mdx",
      development: true,
      root: "xxx",
    });

    expect(formatHTML(html)).toMatchSnapshot();
    expect(result).toMatchSnapshot();
  });

  test("should render container type with space correctly", async (t) => {
    let { code: result, html } = await compile({
      value: readFileSync(
        path.join(__dirname, "./container-type-with-space.md"),
        "utf8"
      ),
      filepath: "xxx.mdx",
      development: true,
      root: "xxx",
    });

    expect(formatHTML(html)).toMatchSnapshot();
    expect(result).toMatchSnapshot();
  });

  test("should render container content correctly", async (t) => {
    let { code: result, html } = await compile({
      value: readFileSync(
        path.join(__dirname, "./container-content.md"),
        "utf8"
      ),
      filepath: "xxx.mdx",
      development: true,
      root: "xxx",
    });

    expect(formatHTML(html)).toMatchSnapshot();
    expect(result).toMatchSnapshot();
  });

  test("should render container title in mdx correctly", async (t) => {
    let { code: result, html } = await compile({
      value: readFileSync(
        path.join(__dirname, "./container-title.mdx"),
        "utf8"
      ),
      filepath: "xxx.mdx",
      development: true,
      root: "xxx",
    });

    expect(formatHTML(html)).toMatchSnapshot();
    expect(result).toMatchSnapshot();
  });

  test("should render container title in md correctly", async (t) => {
    let { code: result, html } = await compile({
      value: readFileSync(path.join(__dirname, "./container-title.md"), "utf8"),
      filepath: "xxx.md",
      development: true,
      root: "xxx",
    });

    expect(formatHTML(html)).toMatchSnapshot();
    expect(result).toMatchSnapshot();
  });

  test("should compile jsx", async (t) => {
    let { code: result, html } = await compile({
      value: readFileSync(path.join(__dirname, "./compile-jsx.mdx"), "utf8"),
      filepath: "xxx.mdx",
      development: true,
      root: "xxx",
      jsx: false
    });

    expect(formatHTML(html)).toMatchSnapshot();
    expect(result).toMatchSnapshot();
  });
});
