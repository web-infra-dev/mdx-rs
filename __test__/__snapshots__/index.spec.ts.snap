// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`compile > should compile jsx 1`] = `
"<h1 id=\\"compile-jsx\\">
  Compile JSX<a aria-hidden=\\"true\\" href=\\"#compile-jsx\\">#</a>
</h1>
"
`;

exports[`compile > should compile jsx 2`] = `
"import { jsxDEV as _jsxDEV } from \\"react/jsx-dev-runtime\\";
import { useMDXComponents as _provideComponents } from \\"@mdx-js/react\\";
function _createMdxContent(props) {
    const _components = Object.assign({
        h1: \\"h1\\",
        a: \\"a\\"
    }, _provideComponents(), props.components);
    return _jsxDEV(_components.h1, {
        id: \\"compile-jsx\\",
        children: [
            \\"Compile JSX\\",
            _jsxDEV(_components.a, {
                className: \\"header-anchor\\",
                \\"aria-hidden\\": \\"true\\",
                href: \\"#compile-jsx\\",
                children: \\"#\\"
            }, undefined, false, {
                fileName: \\"xxx.mdx\\"
            }, this)
        ]
    }, undefined, true, {
        fileName: \\"xxx.mdx\\",
        lineNumber: 1,
        columnNumber: 1
    }, this);
}
function MDXContent(props = {}) {
    const { wrapper: MDXLayout } = Object.assign({}, _provideComponents(), props.components);
    return MDXLayout ? _jsxDEV(MDXLayout, Object.assign({}, props, {
        children: _jsxDEV(_createMdxContent, props, undefined, false, {
            fileName: \\"xxx.mdx\\"
        }, this)
    }), undefined, false, {
        fileName: \\"xxx.mdx\\"
    }, this) : _createMdxContent(props);
}
export default MDXContent;
"
`;

exports[`compile > should render container content correctly 1`] = `
"<div>
  <div>TIP</div>
  <div>
    <p>First line <code>block</code></p>
    <p>Second line <code>block2</code></p>
  </div>
</div>
<div>
  <div>TIP</div>
  <div>
    <ul>
      <li>List item 1 <code>block</code>。</li>
      <li>List item 2 <code>block2</code>。</li>
    </ul>
  </div>
</div>
<div>
  <div>TIP</div>
  <div>
    <p>Some code:</p>
    <pre><code>console.log('hello world');
</code></pre>
  </div>
</div>
"
`;

exports[`compile > should render container content correctly 2`] = `
"import { useMDXComponents as _provideComponents } from \\"@mdx-js/react\\";
function _createMdxContent(props) {
    const _components = Object.assign({
        div: \\"div\\",
        p: \\"p\\",
        code: \\"code\\",
        ul: \\"ul\\",
        li: \\"li\\",
        pre: \\"pre\\"
    }, _provideComponents(), props.components);
    return <><_components.div className=\\"rspress-directive tip\\"><_components.div className=\\"rspress-directive-title\\">{\\"TIP\\"}</_components.div><_components.div className=\\"rspress-directive-content\\"><_components.p>{\\"First line \\"}<_components.code>{\\"block\\"}</_components.code></_components.p>{\\"\\\\n\\"}<_components.p>{\\"Second line \\"}<_components.code>{\\"block2\\"}</_components.code>{\\"\\\\n\\"}</_components.p></_components.div></_components.div>{\\"\\\\n\\"}<_components.div className=\\"rspress-directive tip\\"><_components.div className=\\"rspress-directive-title\\">{\\"TIP\\"}</_components.div><_components.div className=\\"rspress-directive-content\\">{\\"\\\\n\\"}<_components.ul>{\\"\\\\n\\"}<_components.li>{\\"List item 1 \\"}<_components.code>{\\"block\\"}</_components.code>{\\"。\\"}</_components.li>{\\"\\\\n\\"}<_components.li>{\\"List item 2 \\"}<_components.code>{\\"block2\\"}</_components.code>{\\"。\\"}</_components.li>{\\"\\\\n\\"}</_components.ul>{\\"\\\\n\\"}</_components.div></_components.div>{\\"\\\\n\\"}<_components.div className=\\"rspress-directive tip\\"><_components.div className=\\"rspress-directive-title\\">{\\"TIP\\"}</_components.div><_components.div className=\\"rspress-directive-content\\">{\\"\\\\n\\"}<_components.p>{\\"Some code:\\"}</_components.p>{\\"\\\\n\\"}<_components.pre><_components.code className=\\"language-ts\\">{\\"console.log('hello world');\\\\n\\"}</_components.code></_components.pre>{\\"\\\\n\\"}</_components.div></_components.div></>;
}
function MDXContent(props = {}) {
    const { wrapper: MDXLayout } = Object.assign({}, _provideComponents(), props.components);
    return MDXLayout ? <MDXLayout {...props}><_createMdxContent {...props}/></MDXLayout> : _createMdxContent(props);
}
export default MDXContent;
"
`;

exports[`compile > should render container title in md correctly 1`] = `
"<h2 id=\\"custom-title\\">
  Container Title<a aria-hidden=\\"true\\" href=\\"#custom-title\\">#</a>
</h2>
<div>
  <div>Custom Title</div>
  <div>
    <p>This is a <code>block</code> of <code>Custom Title</code></p>
  </div>
</div>
<div>
  <div>Custom Title</div>
  <div>
    <p>This is a <code>block</code> of <code>Custom Title</code></p>
  </div>
</div>
<div>
  <div>Custom Title</div>
  <div>
    <p>This is a <code>block</code> of <code>Custom Title</code></p>
  </div>
</div>
<div>
  <div>Custom Title</div>
  <div>
    <p>This is a <code>block</code> of <code>Custom Title</code></p>
  </div>
</div>
"
`;

exports[`compile > should render container title in md correctly 2`] = `
"import { useMDXComponents as _provideComponents } from \\"@mdx-js/react\\";
function _createMdxContent(props) {
    const _components = Object.assign({
        h2: \\"h2\\",
        a: \\"a\\",
        div: \\"div\\",
        p: \\"p\\",
        code: \\"code\\"
    }, _provideComponents(), props.components);
    return <><_components.h2 id=\\"custom-title\\">{\\"Container Title\\"}<_components.a className=\\"header-anchor\\" aria-hidden=\\"true\\" href=\\"#custom-title\\">{\\"#\\"}</_components.a></_components.h2>{\\"\\\\n\\"}<_components.div className=\\"rspress-directive tip\\"><_components.div className=\\"rspress-directive-title\\">{\\"Custom Title\\"}</_components.div><_components.div className=\\"rspress-directive-content\\"><_components.p>{\\"This is a \\"}<_components.code>{\\"block\\"}</_components.code>{\\" of \\"}<_components.code>{\\"Custom Title\\"}</_components.code>{\\"\\\\n\\"}</_components.p></_components.div></_components.div>{\\"\\\\n\\"}<_components.div className=\\"rspress-directive tip\\"><_components.div className=\\"rspress-directive-title\\">{\\"Custom Title\\"}</_components.div><_components.div className=\\"rspress-directive-content\\"><_components.p>{\\"This is a \\"}<_components.code>{\\"block\\"}</_components.code>{\\" of \\"}<_components.code>{\\"Custom Title\\"}</_components.code>{\\"\\\\n\\"}</_components.p></_components.div></_components.div>{\\"\\\\n\\"}<_components.div className=\\"rspress-directive tip\\"><_components.div className=\\"rspress-directive-title\\">{\\"Custom Title\\"}</_components.div><_components.div className=\\"rspress-directive-content\\"><_components.p>{\\"This is a \\"}<_components.code>{\\"block\\"}</_components.code>{\\" of \\"}<_components.code>{\\"Custom Title\\"}</_components.code>{\\"\\\\n\\"}</_components.p></_components.div></_components.div>{\\"\\\\n\\"}<_components.div className=\\"rspress-directive tip\\"><_components.div className=\\"rspress-directive-title\\">{\\"Custom Title\\"}</_components.div><_components.div className=\\"rspress-directive-content\\"><_components.p>{\\"This is a \\"}<_components.code>{\\"block\\"}</_components.code>{\\" of \\"}<_components.code>{\\"Custom Title\\"}</_components.code>{\\"\\\\n\\"}</_components.p></_components.div></_components.div></>;
}
function MDXContent(props = {}) {
    const { wrapper: MDXLayout } = Object.assign({}, _provideComponents(), props.components);
    return MDXLayout ? <MDXLayout {...props}><_createMdxContent {...props}/></MDXLayout> : _createMdxContent(props);
}
export default MDXContent;
"
`;

exports[`compile > should render container title in mdx correctly 1`] = `
"<h2 id=\\"custom-title\\">
  Container Title <a aria-hidden=\\"true\\" href=\\"#custom-title\\">#</a>
</h2>
<div>
  <div>Custom Title</div>
  <div>
    <p>This is a <code>block</code> of <code>Custom Title</code></p>
  </div>
</div>
<div>
  <div>Custom Title</div>
  <div>
    <p>This is a <code>block</code> of <code>Custom Title</code></p>
  </div>
</div>
<div>
  <div>Custom Title</div>
  <div>
    <p>This is a <code>block</code> of <code>Custom Title</code></p>
  </div>
</div>
<div>
  <div>Custom Title</div>
  <div>
    <p>This is a <code>block</code> of <code>Custom Title</code></p>
  </div>
</div>
"
`;

exports[`compile > should render container title in mdx correctly 2`] = `
"import { useMDXComponents as _provideComponents } from \\"@mdx-js/react\\";
function _createMdxContent(props) {
    const _components = Object.assign({
        h2: \\"h2\\",
        a: \\"a\\",
        div: \\"div\\",
        p: \\"p\\",
        code: \\"code\\"
    }, _provideComponents(), props.components);
    return <><_components.h2 id=\\"custom-title\\">{\\"Container Title \\"}<_components.a className=\\"header-anchor\\" aria-hidden=\\"true\\" href=\\"#custom-title\\">{\\"#\\"}</_components.a></_components.h2>{\\"\\\\n\\"}<_components.div className=\\"rspress-directive tip\\"><_components.div className=\\"rspress-directive-title\\">{\\"Custom Title\\"}</_components.div><_components.div className=\\"rspress-directive-content\\"><_components.p>{\\"\\\\nThis is a \\"}<_components.code>{\\"block\\"}</_components.code>{\\" of \\"}<_components.code>{\\"Custom Title\\"}</_components.code>{\\"\\\\n\\"}</_components.p></_components.div></_components.div>{\\"\\\\n\\"}<_components.div className=\\"rspress-directive tip\\"><_components.div className=\\"rspress-directive-title\\">{\\"Custom Title\\"}</_components.div><_components.div className=\\"rspress-directive-content\\"><_components.p>{\\"\\\\nThis is a \\"}<_components.code>{\\"block\\"}</_components.code>{\\" of \\"}<_components.code>{\\"Custom Title\\"}</_components.code>{\\"\\\\n\\"}</_components.p></_components.div></_components.div>{\\"\\\\n\\"}<_components.div className=\\"rspress-directive tip\\"><_components.div className=\\"rspress-directive-title\\">{\\"Custom Title\\"}</_components.div><_components.div className=\\"rspress-directive-content\\"><_components.p>{\\"This is a \\"}<_components.code>{\\"block\\"}</_components.code>{\\" of \\"}<_components.code>{\\"Custom Title\\"}</_components.code>{\\"\\\\n\\"}</_components.p></_components.div></_components.div>{\\"\\\\n\\"}<_components.div className=\\"rspress-directive tip\\"><_components.div className=\\"rspress-directive-title\\">{\\"Custom Title\\"}</_components.div><_components.div className=\\"rspress-directive-content\\"><_components.p>{\\"This is a \\"}<_components.code>{\\"block\\"}</_components.code>{\\" of \\"}<_components.code>{\\"Custom Title\\"}</_components.code>{\\"\\\\n\\"}</_components.p></_components.div></_components.div></>;
}
function MDXContent(props = {}) {
    const { wrapper: MDXLayout } = Object.assign({}, _provideComponents(), props.components);
    return MDXLayout ? <MDXLayout {...props}><_createMdxContent {...props}/></MDXLayout> : _createMdxContent(props);
}
export default MDXContent;
"
`;

exports[`compile > should render container type correctly 1`] = `
"<div>
  <div>TIP</div>
  <div>
    <p>This is a <code>block</code> of type <code>tip</code></p>
  </div>
</div>
<div>
  <div>INFO</div>
  <div>
    <p>This is a <code>block</code> of type <code>info</code></p>
  </div>
</div>
<div>
  <div>NOTE</div>
  <div>
    <p>This is a <code>block</code> of type <code>info</code></p>
  </div>
</div>
<div>
  <div>WARNING</div>
  <div>
    <p>This is a <code>block</code> of type <code>warning</code></p>
  </div>
</div>
<div>
  <div>DANGER</div>
  <div>
    <p>This is a <code>block</code> of type <code>danger</code></p>
  </div>
</div>
<div>
  <div>CAUTION</div>
  <div>
    <p>This is a <code>block</code> of type <code>danger</code></p>
  </div>
</div>
"
`;

exports[`compile > should render container type correctly 2`] = `
"import { useMDXComponents as _provideComponents } from \\"@mdx-js/react\\";
function _createMdxContent(props) {
    const _components = Object.assign({
        div: \\"div\\",
        p: \\"p\\",
        code: \\"code\\"
    }, _provideComponents(), props.components);
    return <><_components.div className=\\"rspress-directive tip\\"><_components.div className=\\"rspress-directive-title\\">{\\"TIP\\"}</_components.div><_components.div className=\\"rspress-directive-content\\"><_components.p>{\\"This is a \\"}<_components.code>{\\"block\\"}</_components.code>{\\" of type \\"}<_components.code>{\\"tip\\"}</_components.code>{\\"\\\\n\\"}</_components.p></_components.div></_components.div>{\\"\\\\n\\"}<_components.div className=\\"rspress-directive info\\"><_components.div className=\\"rspress-directive-title\\">{\\"INFO\\"}</_components.div><_components.div className=\\"rspress-directive-content\\"><_components.p>{\\"This is a \\"}<_components.code>{\\"block\\"}</_components.code>{\\" of type \\"}<_components.code>{\\"info\\"}</_components.code>{\\"\\\\n\\"}</_components.p></_components.div></_components.div>{\\"\\\\n\\"}<_components.div className=\\"rspress-directive note\\"><_components.div className=\\"rspress-directive-title\\">{\\"NOTE\\"}</_components.div><_components.div className=\\"rspress-directive-content\\"><_components.p>{\\"This is a \\"}<_components.code>{\\"block\\"}</_components.code>{\\" of type \\"}<_components.code>{\\"info\\"}</_components.code>{\\"\\\\n\\"}</_components.p></_components.div></_components.div>{\\"\\\\n\\"}<_components.div className=\\"rspress-directive warning\\"><_components.div className=\\"rspress-directive-title\\">{\\"WARNING\\"}</_components.div><_components.div className=\\"rspress-directive-content\\"><_components.p>{\\"This is a \\"}<_components.code>{\\"block\\"}</_components.code>{\\" of type \\"}<_components.code>{\\"warning\\"}</_components.code>{\\"\\\\n\\"}</_components.p></_components.div></_components.div>{\\"\\\\n\\"}<_components.div className=\\"rspress-directive danger\\"><_components.div className=\\"rspress-directive-title\\">{\\"DANGER\\"}</_components.div><_components.div className=\\"rspress-directive-content\\"><_components.p>{\\"This is a \\"}<_components.code>{\\"block\\"}</_components.code>{\\" of type \\"}<_components.code>{\\"danger\\"}</_components.code>{\\"\\\\n\\"}</_components.p></_components.div></_components.div>{\\"\\\\n\\"}<_components.div className=\\"rspress-directive caution\\"><_components.div className=\\"rspress-directive-title\\">{\\"CAUTION\\"}</_components.div><_components.div className=\\"rspress-directive-content\\"><_components.p>{\\"This is a \\"}<_components.code>{\\"block\\"}</_components.code>{\\" of type \\"}<_components.code>{\\"danger\\"}</_components.code>{\\"\\\\n\\"}</_components.p></_components.div></_components.div></>;
}
function MDXContent(props = {}) {
    const { wrapper: MDXLayout } = Object.assign({}, _provideComponents(), props.components);
    return MDXLayout ? <MDXLayout {...props}><_createMdxContent {...props}/></MDXLayout> : _createMdxContent(props);
}
export default MDXContent;
"
`;

exports[`compile > should render container type with space correctly 1`] = `
"<div>
  <div>TIP</div>
  <div>
    <p>This is a <code>block</code> of type <code>tip</code></p>
  </div>
</div>
<div>
  <div>TIP</div>
  <div>
    <p>This is a <code>block</code> of type <code>tip</code></p>
  </div>
</div>
<div>
  <div>INFO</div>
  <div>
    <p>This is a <code>block</code> of type <code>info</code></p>
  </div>
</div>
<div>
  <div>NOTE</div>
  <div>
    <p>This is a <code>block</code> of type <code>info</code></p>
  </div>
</div>
<div>
  <div>WARNING</div>
  <div>
    <p>This is a <code>block</code> of type <code>warning</code></p>
  </div>
</div>
<div>
  <div>DANGER</div>
  <div>
    <p>This is a <code>block</code> of type <code>danger</code></p>
  </div>
</div>
<div>
  <div>CAUTION</div>
  <div>
    <p>This is a <code>block</code> of type <code>danger</code></p>
  </div>
</div>
"
`;

exports[`compile > should render container type with space correctly 2`] = `
"import { useMDXComponents as _provideComponents } from \\"@mdx-js/react\\";
function _createMdxContent(props) {
    const _components = Object.assign({
        div: \\"div\\",
        p: \\"p\\",
        code: \\"code\\"
    }, _provideComponents(), props.components);
    return <><_components.div className=\\"rspress-directive tip\\"><_components.div className=\\"rspress-directive-title\\">{\\"TIP\\"}</_components.div><_components.div className=\\"rspress-directive-content\\"><_components.p>{\\"This is a \\"}<_components.code>{\\"block\\"}</_components.code>{\\" of type \\"}<_components.code>{\\"tip\\"}</_components.code>{\\"\\\\n\\"}</_components.p></_components.div></_components.div>{\\"\\\\n\\"}<_components.div className=\\"rspress-directive tip\\"><_components.div className=\\"rspress-directive-title\\">{\\"TIP\\"}</_components.div><_components.div className=\\"rspress-directive-content\\"><_components.p>{\\"This is a \\"}<_components.code>{\\"block\\"}</_components.code>{\\" of type \\"}<_components.code>{\\"tip\\"}</_components.code>{\\"\\\\n\\"}</_components.p></_components.div></_components.div>{\\"\\\\n\\"}<_components.div className=\\"rspress-directive info\\"><_components.div className=\\"rspress-directive-title\\">{\\"INFO\\"}</_components.div><_components.div className=\\"rspress-directive-content\\"><_components.p>{\\"This is a \\"}<_components.code>{\\"block\\"}</_components.code>{\\" of type \\"}<_components.code>{\\"info\\"}</_components.code>{\\"\\\\n\\"}</_components.p></_components.div></_components.div>{\\"\\\\n\\"}<_components.div className=\\"rspress-directive note\\"><_components.div className=\\"rspress-directive-title\\">{\\"NOTE\\"}</_components.div><_components.div className=\\"rspress-directive-content\\"><_components.p>{\\"This is a \\"}<_components.code>{\\"block\\"}</_components.code>{\\" of type \\"}<_components.code>{\\"info\\"}</_components.code>{\\"\\\\n\\"}</_components.p></_components.div></_components.div>{\\"\\\\n\\"}<_components.div className=\\"rspress-directive warning\\"><_components.div className=\\"rspress-directive-title\\">{\\"WARNING\\"}</_components.div><_components.div className=\\"rspress-directive-content\\"><_components.p>{\\"This is a \\"}<_components.code>{\\"block\\"}</_components.code>{\\" of type \\"}<_components.code>{\\"warning\\"}</_components.code>{\\"\\\\n\\"}</_components.p></_components.div></_components.div>{\\"\\\\n\\"}<_components.div className=\\"rspress-directive danger\\"><_components.div className=\\"rspress-directive-title\\">{\\"DANGER\\"}</_components.div><_components.div className=\\"rspress-directive-content\\"><_components.p>{\\"This is a \\"}<_components.code>{\\"block\\"}</_components.code>{\\" of type \\"}<_components.code>{\\"danger\\"}</_components.code>{\\"\\\\n\\"}</_components.p></_components.div></_components.div>{\\"\\\\n\\"}<_components.div className=\\"rspress-directive caution\\"><_components.div className=\\"rspress-directive-title\\">{\\"CAUTION\\"}</_components.div><_components.div className=\\"rspress-directive-content\\"><_components.p>{\\"This is a \\"}<_components.code>{\\"block\\"}</_components.code>{\\" of type \\"}<_components.code>{\\"danger\\"}</_components.code>{\\"\\\\n\\"}</_components.p></_components.div></_components.div></>;
}
function MDXContent(props = {}) {
    const { wrapper: MDXLayout } = Object.assign({}, _provideComponents(), props.components);
    return MDXLayout ? <MDXLayout {...props}><_createMdxContent {...props}/></MDXLayout> : _createMdxContent(props);
}
export default MDXContent;
"
`;
